lib/internal/fs/utils.js

```diff
@@ -291,8 +291,10 @@ function validateBuffer(buffer) {
 
 function validateOffsetLengthRead(offset, length, bufferLength) {
   let err;
-
-  if (offset < 0 || offset >= bufferLength) {
+  if (bufferLength === 0) {
+    err = new ERR_INVALID_ARG_VALUE('buffer', 0,
+                                    "is empty and can't be written.");
```

Comment:
By the way, in this code base we use single quotes. Please run `make lint` (non-Windows) or `vcbuild.bat lint` (Windows) to run the tests.

Reply:
Please also move this statement in the already existing if statement and change it to an `ERR_OUT_OF_RANGE` error. That way the extra check is only triggered in the error case as in:

```js
if (offset < 0 || offset >= bufferLength) {
  if (bufferLength === 0) {
    err = new ERR_OUT_OF_RANGE('bufferLength', '> 0', bufferLength);
  } else {
    err = new ...;
  }
} else ...
```